components/tidb_query_expr/src/impl_math.rs

```diff
@@ -460,6 +483,28 @@ pub fn truncate_int_with_uint(arg0: &Int, _arg1: &Int) -> Result<Option<Int>> {
     Ok(Some(*arg0))
 }
 
+#[inline]
+#[rpn_fn]
+pub fn truncate_uint_with_int(arg0: &Int, arg1: &Int) -> Result<Option<Int>> {
+    let x = arg0;
+    let d = arg1;
+    Ok(Some(if *d >= 0 {
+        *x
+    } else if *d <= -(I64_TEN_POWS.len() as i64) {
```

Comment:
Logic in tidb:

```go
	if d == mathutil.MinInt {
		return 0, false, nil
	}
```

Reply:
An `u64` is between `[0, 1 << 64)`, which means an `u64` is not longer than `20` digits.

Thus we can just take at most `20` digits in consideration. This is the reason of the original logic in `tikv`.

See also [here](https://github.com/tikv/tikv/blob/1a88f12ebf50064b992fc5efc7e7f55795210521/components/tidb_query_normal_expr/src/builtin_math.rs#L424)

Reply:
Maybe we can introduce a constant like something called `MAX_DIGIT_LENGTH` instead of depends on `I64_TEN_POWS.len()` ?

Reply:
okay, let me remove it.

done